### PR TITLE
fix: enabling semisync; feat: errno in logs

### PR DESCRIPTION
**Pull request by Fizic**

### CODE REVIEW

File: internal/app/app.go [hunk 1]
```diff
@@ -320,7 +320,12 @@ func (app *App) checkRecovery() {
 	app.logger.Infof("recovery: local node %s has GTIDs %s", localNode.Host(), sstatus.GetExecutedGtidSet())
 
 	if isSlavePermanentlyLost(sstatus, mgtids) {
-		app.logger.Errorf("recovery: local node %s is NOT behind the master %s, need RESETUP", localNode.Host(), masterNode)
+		rp, err := localNode.GetReplicaStatus()
+		if err != nil {
+			app.logger.Errorf("recovery: local node %s is NOT behind the master %s, need RESETUP", localNode.Host(), masterNode)
```
Comment (moridin26) [inline internal/app/app.go:325]: you probably should remove phrase about "...NOT behind the master... "

File: internal/app/app.go [hunk 2]
```diff
@@ -320,7 +320,12 @@ func (app *App) checkRecovery() {
...
+		} else {
+			app.logger.Errorf("recovery: local node %s is NOT behind the master %s, need RESETUP, last IO errno: %d", localNode.Host(), masterNode, rp.GetLastIOErrno())
```
Comment (moridin26) [inline internal/app/app.go:327]: it seems that you should switch "if" and "else" blocks

File: internal/app/app.go [hunk 3]
```diff
@@ -1072,6 +1077,11 @@ func (app *App) enableSemiSyncOnSlave(host string) error {
 		app.logger.Errorf("failed restart slave io thread after set semi_sync_slave on %s: %s", host, err)
 		return err
 	}
+	err = node.RestartReplication()
```
Comment (moridin26) [inline internal/app/app.go:1080]: we should probably remove call `node.RestartSlaveIOThread()` from code above

File: internal/mysql/node.go [hunk 4]
```diff
@@ -722,6 +722,15 @@ func (n *Node) StartSlave() error {
 	})
 }
 
+// RestartReplication restart replication
+func (n *Node) RestartReplication() error {
```
Comment (teem0n) [inline internal/mysql/node.go:726]: Let's rename this to 'RestartReplica'

**Review by teem0n — APPROVED** [review]

**Pull request merged**